src/adapter/auth-web3/tests/tests/test_web3_wallet_authentication_provider.rs

```diff
@@ -0,0 +1,390 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::assert_matches::assert_matches;
+use std::sync::{Arc, LazyLock};
+
+use chrono::{TimeZone, Utc};
+use kamu_accounts::{AuthenticationProvider, ProviderLoginError};
+use kamu_adapter_auth_web3::{LoginRequest, Web3WalletAuthenticationProvider};
+use kamu_auth_web3::{EvmWalletAddressConvertor, Web3AuthEip4361NonceService};
+use kamu_auth_web3_inmem::InMemoryWeb3AuthEip4361NonceRepository;
+use kamu_auth_web3_services::Web3AuthEip4361NonceServiceImpl;
+use kamu_core::ServerUrlConfig;
+use serde_json::json;
+use time_source::{SystemTimeSource, SystemTimeSourceStub};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+const WALLET_ADDRESS: &str = "0x90C25A5947c24275eDC4544b74Aa11864F79eeAF";
```

Comment:
What is this address? Hope you did not enter your wallet.
Is there any wallet address in ETH world that can act like @example.com domain for URLs?

Reply:
> What is this address? Hope you did not enter your wallet.

One of the test accounts I had with MetaMask -- but I made another one anyway, just for this test

> Is there any wallet address in ETH world that can act like @example.com domain for URLs?

Unfortunately, nope